### PR TITLE
add resource limits and requests

### PR DESCRIPTION
**Pull request by dashanji**
The PR is related to https://github.com/apache/skywalking/issues/8258.

### CODE REVIEW

File: operator/pkg/operator/injector/injector.go [hunk 1]
```diff
@@ -252,10 +258,32 @@ func (s *SidecarInjectField) OverlaySidecar(a Annotations, ao *AnnotationOverlay
...
 
+	// add requests and limits to initcontainer
+	if limitsStr != "nil" {
+		limits := make(corev1.ResourceList)
+		err := json.Unmarshal([]byte(limitsStr), &limits)
+		if err != nil {
+			log.Error(err, "unmarshal limitsStr error")
+			return false
+		}
+		s.Initcontainer.Resources.Limits = limits
```
Comment (hanahmily) [inline operator/pkg/operator/injector/injector.go:274]: Would you please validate the string by a checker?

**Review by hanahmily — CHANGES_REQUESTED** [review]
Could you add the resources request/limit to one of the existing e2e test cases?

File: operator/pkg/operator/injector/injector.go (hunk 1 above)
Comment (dashanji) [reply operator/pkg/operator/injector/injector.go:274]: OK, get it.

**Review by hanahmily — APPROVED** [review]
LGTM

**Pull request merged**